packages/controller/src/main.ts

```diff
@@ -3132,6 +3129,40 @@ async function processMessage(msg: ioBroker.SendableMessage): Promise<null | voi
             break;
         }
 
+        case 'upgradeController': {
+            if (['win32', 'darwin'].includes(os.platform())) {
```

Comment:
can we also detect "non-systemd variants?"

Reply:
We don’t have this detection yet, if needed we can look into it, there sure will be a way. 

However, as we are calling `iob stop/start` it should then be handled there and not in this place if adaption is needed